### PR TITLE
Add at rest encryption to Meterpreter payloads

### PR DESCRIPTION
**Pull request by sjanusz-r7**
This PR is requires https://github.com/rapid7/metasploit-payloads/pull/679
The change to `client_core.rb` is necessary to support encrypted stdapi libraries.
Other changes were made to support encrypted file contents being added to zip/jar/archive files by using `::MetasploitPayloads.read` which will handle encrypted/plain-text payloads as `#add_files` cannot take in a buffer, and we have to resort to calling `#add_file` for each file with the contents instead.

## Verification

- [ ] Start `msfconsole`
- [ ] Use a payload, for example `use payload/python/meterpreter/reverse_tcp`
- [ ] Ensure you can get get a session as usual
- [ ] When in `irb`, check that the payload being read is stored as an encrypted file on disk
  - [ ] `path = ::MetasploitPayloads.path('meterpreter', 'meterpreter.py')`
  - [ ] `raw_contents = ::File.binread(path); nil`
  - [ ] `raw_contents[0..5] == "msf\x01\x01\x01"`
  - [ ] `decrypted_contents = ::MetasploitPayloads.read('meterpreter', 'meterpreter.py')`

### CODE REVIEW

**Comment by sempervictus** [comment]
Is there some place where we can find a bit of conext for the intended use of this functionality

**Comment by adfoster-r7** [comment]
The title was a bit of a misnomer. We're just adding at rest encryption for the Meterpreter payloads on disk, the changes should be transparent for the average user

**Comment by sempervictus** [comment]
Pesky AVs not appreciating users having the files on their machines?
Might be interesting to have runtime decryption on the other side for conditional loading/clearing of extensions when not in use to an encrypted read-only "inert code cave" (or just from remote again) for reloading on-demand. Might help reduce our runtime fingerprints in-memory.

**Comment by sempervictus** [comment]
Is there a facility for dumping the raw DLLs? Other tools might want access to them...

File: modules/exploits/windows/local/ms15_078_atmfd_bof.rb [hunk 1]
```diff
@@ -384,8 +384,8 @@ def exploit
     library_path = ::File.expand_path(library_path)
 
     print_status("Reflectively injecting the exploit DLL into #{process.pid}...")
-    dll = ''
-    ::File.open(library_path, 'rb') { |f| dll = f.read }
+    encrypted_dll = ::File.binread(library_path)
+    dll = ::MetasploitPayloads::Crypto.decrypt(ciphertext: encrypted_dll)
```
Comment (adfoster-r7) [inline modules/exploits/windows/local/ms15_078_atmfd_bof.rb:388]:
```suggestion
    dll = ::File.binread(library_path)
```

**Comment by adfoster-r7** [comment]
## Release Notes

Add at rest encryption to Meterpreter payloads on the Metasploit host machine's file system

**Pull request merged**